app/javascript/components/download/FileDownloadControl.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import { fetchBucketFile } from 'lib/scp-api'
+import { bytesToSize } from 'lib/stats'
+
+/** renders a file download control for the given file object */
+export default function FileDownloadControl({ file, bucketName }) {
+  const fileName = file.upload_file_name
+
+  /** Load file when button is clicked */
+  const handleOnClick = () => {
+    getBucketLocalUrl(bucketName, fileName).then(value => window.open(value))
```

Comment:
for clarity, and since you declared the `getBucketLocalUrl` function async
```suggestion
    const localUrl = await getBucketLocalUrl(bucketName, fileName)
    window.open(url)
```

Reply:
Also, since this is opening an arbitrary file, we should be careful about security.  Reading https://stackoverflow.com/questions/40593632/use-window-open-but-block-use-of-window-opener, it seems like the arguments below are the equivalent of `rel="noopener noreferrer"` 
`window.open(url, '_blank', 'noopener,noreferrer)`

Reply:
I actually find the `then` structure more clear in this case. Since `getBucketLocalUrl` returns a promise the `then` just consume the value returned when the promise fulfills and opens the window using that value which feels simpler than having to await and make `handleDownloadClick` `async` and then needing to `await` and `async` the `onclick` in the <a> as well. Please let me know though if you feel strongly about this and I can be persuaded haha.